compiler/rustc_lint/src/lib.rs

```diff
@@ -303,19 +303,25 @@ fn register_builtins(store: &mut LintStore) {
         BARE_TRAIT_OBJECTS,
         UNUSED_EXTERN_CRATES,
         ELLIPSIS_INCLUSIVE_RANGE_PATTERNS,
-        ELIDED_LIFETIMES_IN_PATHS,
-        EXPLICIT_OUTLIVES_REQUIREMENTS // FIXME(#52665, #47816) not always applicable and not all
-                                       // macros are ready for this yet.
-                                       // UNREACHABLE_PUB,
+        ELIDED_LIFETIMES_IN_PATHS_TIED,
+        ELIDED_LIFETIMES_IN_PATHS_UNTIED,
```

Comment:
If we are to nitpick about this, an _elided_ lifetime refers to both `Foo` and `Foo<'_>` (as per the official lifetime elision rules), with the former being implicitly elided, and the latter, explicitly (very unfortunate that the original lint picked that name 😔). _Hidden_ is a more concise word which we could reach for, now.

For the purpose of the lint, I'd also move the adjective around, since `elided_lifetimes_in_paths_tied` does not roll off the tongue too much.

Finally, on the most controversial/debatable aspect of the exact word being picked here (_e.g._, "tied"), my own subjective two cents on the topic, would be to not overthink it, and use _meaningful_: `meaningful_lifetimes_hidden_in_paths`. The other one could use _free_: `free_lifetimes_hidden_in_paths`.

Reply:
I'm sympathetic to this feedback, but some points arise when trying to implement some of the suggestions:

- Calling the lints `a_commonpart` and `b_commonpart` means that they won't appear near each other in alphabetical lists (which are numerous).
- There's another lint (`ELIDED_LIFETIMES_IN_ASSOCIATED_CONSTANT`) that should probably be updated.
- I'm torn between "hidden lifetimes" and "lifetimes hidden".
- There's a lot of other "elided" lifetimes in `rustc_resolve`'s `late.rs`. I don't know which style they mean. Having both terms with mixed meanings may make things worse until someone cleans up all the code.

Reply:
Yeah, fair enough

Reply:
...and I removed that commit from _this_ PR, although I still have it locally. It looks like this PR will have enough trouble getting feedback, so let's go smaller.

Reply:
When it comes to nomenclature of the lints, I have a bias towards keeping the name of the group as a substring of the individual lints, and I agree with some of daniel's observations, so my personal preference would be to name these lints `ELIDED_LIFETIMES_IN_PATHS_EXPLICIT` (not 100% sure on this one) and `ELIDED_LIFETIMES_IN_PATHS_HIDDEN`.